lib/internal/streams/pipeline.js

```diff
@@ -1,97 +1,416 @@
 // Ported from https://github.com/mafintosh/pump with
 // permission from the author, Mathias Buus (@mafintosh).
-'use strict';
+'use strict'
 
-var eos;
+const { ArrayIsArray, Promise, SymbolAsyncIterator } = require('../../ours/primordials')
 
-function once(callback) {
-  var called = false;
-  return function () {
-    if (called) return;
-    called = true;
-    callback.apply(void 0, arguments);
-  };
+const eos = require('./end-of-stream')
+
+const { once } = require('../../ours/util')
+
+const destroyImpl = require('./destroy')
+
+const Duplex = require('./duplex')
+
+const {
+  aggregateTwoErrors,
+  codes: { ERR_INVALID_ARG_TYPE, ERR_INVALID_RETURN_VALUE, ERR_MISSING_ARGS, ERR_STREAM_DESTROYED },
+  AbortError
+} = require('../../ours/errors')
+
+const { validateFunction, validateAbortSignal } = require('../validators')
+
+const { isIterable, isReadable, isReadableNodeStream, isNodeStream } = require('./utils')
+
+if (typeof AbortController === 'undefined') {
+  globalThis.AbortController = require('abort-controller').AbortController
```

Comment:
We can't add globals as a side effect of loading readable-stream

Reply:
👍